### ISSUE TITLE
[feat] `alwaysOnTop` support for windows above the dock on macOS and above the taskbar on Windows

### ISSUE THREAD

**Issue by ayangweb** [tauri-apps/tauri#9987]
### Describe the problem

I don't know if tauri currently has a way to make it possible for windows to show above the Dock on macOS and above the taskbar on Windows, if not! Is there a quick way to make it happen?

### Describe the solution you'd like

electron's [setAlwaysOnTop](https://www.electronjs.org/zh/docs/latest/api/browser-window#winsetalwaysontopflag-level-relativelevel) Support for windows above the dock on macOS and above the taskbar on Windows.

**Comment by ayangweb** [comment]
@FabianLars Hey 👋, is there any progress on this issue? I see you reopened it.

On windows, the window is right above the taskbar, so there's no need to do anything else!

**Comment by ayangweb** [comment]
macos This is how I currently implement it!

```rs
unsafe {
    use cocoa::appkit::{NSMainMenuWindowLevel, NSWindow};
    use cocoa::base::id;

    let ns_window = main_window.ns_window().unwrap() as id;

    ns_window.setLevel_(NSMainMenuWindowLevel as i64 + 1);
}
```

**Comment by Waradu** [comment]
> On windows, the window is right above the taskbar, so there's no need to do anything else!

As soon as I click on the taskbar or press the windows key the window goes behind the taskbar (Win11).
I have set "alwaysOnTop" to true in my tauri.conf.json

**Comment by FabianLars** [comment]
> is there any progress on this issue? I see you reopened it.

Nope. Just came up on discord and this is the only relevant tracking issue i found.
